### ISSUE TITLE
SDL2 compatible libraries

### ISSUE THREAD

**Issue by Rydgel** [haskell-game/sdl2#79]
_This is not an issue, I'm just unsure where to post this. Feel free to move it._

First of all, congratulations on the release of **sdl2-2.0.0**. We have some really nice improvements here.

I would like to talk about the state of the various SDL2 sub-libraries.
So far the most used ones are:
 
 + sdl2_image
 + sdl2_ttf
 + sdl2_mixer
 + sdl2_net
 + sdl2_gfx

There are existing wrappers for those one. Some are not on Hackage, some are but relies on sdl2-1.3.1.
And so on. Some work have been started, for example by [@sbidin](https://github.com/sbidin?tab=repositories), he updated his libs to work with the current version of **sdl2**.

I'm not sure what is the best way to deal with this, like contacting old library mainteners or starting doing them on our own. They should have the same treatment that you guys did for the sdl2 one. I think it would be awesome to have those wrappers inside `haskell-game` and Hackage.

Please tell me what do you guys think about this.

I am willing to help doing them.

**Comment by ocharles** [comment]
> I think it would be awesome to have those wrappers inside haskell-game and Hackage.

I agree, I would be more than happy to incorporate those wrappers into this repository in fact. I can imagine each wrapper having a hierarchy like:

```
SDL.Image
SDL.Image.Raw
```

where `SDL.Image` is a nice Haskell API and `SDL.Image.Raw` is the underlying FFI representation.

I wouldn't fold these libraries directly into the `sdl2` project, mostly because it complicates distribution of the C libraries. However, having them versioned in this repository makes it easy to `git grep` for API changes, and gives us a nice central place to track issues (that may be cross-cutting concerns).

**Comment by Rydgel** [comment]
I will be working on sdl2_image this weekend. I will study your high-levels bindings to make something similar and compatible. And working on small examples :)

**Comment by ocharles** [comment]
Fantastic! I suggest working in a fork of this repository, and creating an `sdl2-image` directory there.

**Comment by sdroege** [comment]
Might also make sense to import the others to give them more visibility and maybe attract more people willing to work on them

**Comment by Rydgel** [comment]
Sorry people, I'm busier than expected. I couldn't start working on this.

**Comment by sbidin** [comment]
Aside from `sdl2_net` (which is commonly not wrapped at all due to the language providing better networking mechanisms out of the box), the libraries mentioned in the introductory post are now all on Hackage and I believe compatible with the latest haskell-game/sdl2 version:

* http://hackage.haskell.org/package/sdl2-image
* http://hackage.haskell.org/package/sdl2-ttf
* http://hackage.haskell.org/package/sdl2-mixer
* http://hackage.haskell.org/package/sdl2-gfx

**Comment by Rydgel** [comment]
Thank you this is awesome @sbidin

**Comment by tejon** [comment]
@sbidin Any chance you'd like to [add those to Stackage](https://github.com/fpco/stackage/blob/master/MAINTAINERS.md#adding-a-package)?

**Comment by sbidin** [comment]
Sure!

I've opened [a pull request](https://github.com/fpco/stackage/pull/2264) to do so for all libs except `sdl2-ttf`, which I am not a maintainer of.

**Comment by rongcuid** [comment]
I am glad to, but how do I add? Also, I am not entirely sure whether my package is stable enough, because I have not touched it recently, so it would be quite insufficiently tested.

**Comment by tejon** [comment]
[How to add.](https://github.com/fpco/stackage/blob/master/MAINTAINERS.md#adding-a-package)

As long as it builds within the current Stackage Nightly, I think it's fine; Stackage isn't concerned with what the package does, only that it's fully dependency-compatible within the environment. Worst case, you probably get some more eyes finding bugs!

**Comment by sbidin** [comment]
[Stackage LTS 8.0](https://www.stackage.org/diff/lts-7.19/lts-8.0) is now out, and it includes `sdl2-image`, `sdl2-mixer` and `sdl2-gfx`.

**Comment by rongcuid** [comment]
Give me some time. I am hit by numerous deadlines these days

**Comment by juhp** [comment]
Thanks

I think this needa the SDL_ttf C library to build?

**Comment by rongcuid** [comment]
Yes it does

**Comment by juhp** [comment]
Thanks  (Sorry I meant to comment in the stackage ticket.:)

**Comment by sbidin** [comment]
Is there still interest in adding the rest of the sdl2 libraries to haskell-game? If so, I'd love to help migrate sdl2-image, sdl2-mixer and sdl2-gfx over here, adding any changes you deem necessary.

**Comment by ocharles** [comment]
@sbidin certainly, I think all you have to do is transfer them to the haskell-game organisation.

**Comment by sbidin** [comment]
@ocharles Great, I just need permission to do so. I get a "You don’t have the permission to create repositories on haskell-game" at the moment. I think an org admin needs to grant permission explicitly somehow?

**Comment by ocharles** [comment]
Ok, let me look into that.

**Comment by sbidin** [comment]
@ocharles Any update? I'll soon be archiving my repos because I don't have time to attend to them, so if wanted one can clone them into haskell-game at a later date.

**Comment by sbidin** [comment]
@ocharles Ok, you've added me to the organization. I cannot transfer ownership of the repos however, until I'm granted "permission to create repositories on haskell-game".

**Comment by ocharles** [comment]
Alright, let me see if I can figure that bit out.

**Comment by ocharles** [comment]
Try now?

**Comment by sbidin** [comment]
Works! I'll transfer them now.

**Comment by ocharles** [comment]
Woohoo! Thanks, sorry it took me so long.
